export/index.html

```diff
@@ -268,12 +282,120 @@ <h2>Message log</h2>
+        }
+
+        // Add '1' for each leading 0 byte
+        for (let i = 0; bytes[i] === 0 && i < bytes.length - 1; i++) {
+            result = '1' + result;
+        }
+        return result;
+      }
+
+      /**
+       * Decodes a base58-encoded string into a buffer
+       * This function throws an error when the string contains invalid characters.
+       * @param {string} s The base58-encoded string.
+       * @return {Uint8Array} The decoded buffer.
+       */
+       function base58Decode(s) {
+        // See https://en.bitcoin.it/wiki/Base58Check_encoding
+        var alphabet = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+        var decoded = BigInt(0);
+        var decodedBytes = [];
+        var leadingZeros = [];
+        for (var i = 0; i < s.length; i++) {
+          if (alphabet.indexOf(s[i]) === -1) {
+            throw new Error(`cannot base58-decode: ${s[i]} isn't a valid character`)
+          }
+          var carry = alphabet.indexOf(s[i]);
+
+          // If the current base58 digit is 0, append a 0 byte.
+          // "i == leadingZeros.length" can only be true if we have not seen non-zero bytes so far.
+          // If we had seen a non-zero byte, carry wouldn't be 0, and i would be strictly more than `leadingZeros.length`
+          if (carry == 0 && i === leadingZeros.length) {
+            leadingZeros.push(0);
+          }
+
+          var j = 0;
+          while (j < decodedBytes.length || carry > 0) {
+            var currentByte = decodedBytes[j];
+
+            // shift the current byte 58 units and add the carry amount
+            // (or just add the carry amount if this is a new byte -- undefined case)
+            if (currentByte === undefined) {
+              currentByte = carry
+            } else {
+              currentByte = currentByte * 58 + carry
+            }
+
+            // find the new carry amount (1-byte shift of current byte value)
+            carry = currentByte >> 8;
+            // reset the current byte to the remainder (the carry amount will pass on the overflow)
+            decodedBytes[j] = currentByte % 256;
+            j++
+          }
+        }
+
+        var result = leadingZeros.concat(decodedBytes.reverse());
+        return new Uint8Array(result);
+      }
+
+      /**
+       * Returns a private key from private key bytes, represented in
+       * the encoding and format specified by `keyFormat`. Defaults to
+       * hex-encoding if `keyFormat` isn't passed.
        * @param {Uint8Array} privateKeyBytes
+       * @param {string} keyFormat Can be "HEXADECIMAL" or "SOLANA"
+       * @param {string} publicKey Hex-encoded public key, needed for Solana private keys
        */
-      const parseKey = privateKeyBytes => {
-        const privateKeyHexString = uint8arrayToHexString(privateKeyBytes);
-        return "0x" + privateKeyHexString;
+      const parseKey = (privateKeyBytes, keyFormat, publicKey) => {
```

Comment:
This isn't a parsing (text => bytes) function, it's the other way around! We go from bytes to text. So I think this should be "encodeKey" or "encodePrivateKey"?